engine/services/download_service.cc

```diff
@@ -276,16 +218,196 @@ curl_off_t DownloadService::GetLocalFileSize(
   return file_size;
 }
 
-void DownloadService::CleanUp(const std::string& task_id) {
-  CTL_INF("Cleaning up download task: " << task_id);
-  // TODO: might need to be wrap with mutex
-  // remove from task list
-  download_task_list_.erase(std::remove(download_task_list_.begin(),
-                                        download_task_list_.end(), task_id),
-                            download_task_list_.end());
-  // remove from task map
-  download_task_map_.erase(task_id);
-
-  active_download_task_id_ = std::nullopt;
-  active_download_item_id_ = std::nullopt;
+void DownloadService::WorkerThread() {
+  while (!stop_flag_) {
+    DownloadTask task;
+    {
+      std::unique_lock<std::mutex> lock(queue_mutex_);
+      queue_condition_.wait(
+          lock, [this] { return !task_queue_.empty() || stop_flag_; });
+      if (stop_flag_) {
+        CTL_INF("Stopping download service..");
+        break;
+      }
+      if (!task_queue_.empty()) {
```

Comment:
Not sure why we need to check empty() here?

Reply:
Good catch! It's unnecessary after I clean up that function. Removed.